packages/react-instantsearch-dom/src/components/VoiceSearch.tsx

```diff
@@ -24,41 +24,30 @@ type VoiceSearchProps = {
   statusComponent?: React.FC<InnerComponentProps>;
 };
 
-const getDefaultButtonInnerElement = (
-  status: string,
-  errorCode: string | undefined,
-  isListening: boolean
-) => {
-  if (status === 'error' && errorCode === 'not-allowed') {
-    return (
-      <>
-        <line x1="1" y1="1" x2="23" y2="23" />
-        <path d="M9 9v3a3 3 0 0 0 5.12 2.12M15 9.34V4a3 3 0 0 0-5.94-.6" />
-        <path d="M17 16.95A7 7 0 0 1 5 12v-2m14 0v2a7 7 0 0 1-.11 1.23" />
-        <line x1="12" y1="19" x2="12" y2="23" />
-        <line x1="8" y1="23" x2="16" y2="23" />
-      </>
-    );
-  }
-  return (
-    <>
-      <path
-        d="M12 1a3 3 0 0 0-3 3v8a3 3 0 0 0 6 0V4a3 3 0 0 0-3-3z"
-        fill={isListening ? 'currentColor' : ''}
-      />
-      <path d="M19 10v2a7 7 0 0 1-14 0v-2" />
-      <line x1="12" y1="19" x2="12" y2="23" />
-      <line x1="8" y1="23" x2="16" y2="23" />
-    </>
-  );
-};
-
 const DefaultButton: React.FC<InnerComponentProps> = ({
   status,
   errorCode,
   isListening,
 }) => {
-  return (
+  return status === 'error' && errorCode === 'not-allowed' ? (
+    <svg
```

Comment:
You could create a component for both `svg` element, it would help to understand what is rendered in which condition.

Reply:
https://github.com/algolia/react-instantsearch/pull/2316/commits/48d6dc4e4b479ad8fc418183112e284f91c9484d